src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.ts

```diff
@@ -120,63 +130,136 @@ export class StudentExamDetailComponent implements OnInit {
+        }
     }
 
+    /**
+     * Updates the form values based on the working time of the student exam.
+     * @private
+     */
     private initWorkingTimeForm() {
-        const workingTime = this.artemisDurationFromSecondsPipe.secondsToDuration(this.studentExam.workingTime!);
-        this.workingTimeForm = new FormGroup({
-            hours: new FormControl({ value: workingTime.days * 24 + workingTime.hours, disabled: this.examIsVisible() }, [Validators.min(0), Validators.required]),
-            minutes: new FormControl({ value: workingTime.minutes, disabled: this.examIsVisible() }, [Validators.min(0), Validators.max(59), Validators.required]),
-            seconds: new FormControl({ value: workingTime.seconds, disabled: this.examIsVisible() }, [Validators.min(0), Validators.max(59), Validators.required]),
-        });
+        this.setWorkingTimeDuration(this.studentExam.workingTime!);
+        this.updateWorkingTimePercent();
+    }
+
+    /**
+     * Updates the working time duration values of the form whenever the percent value has been changed by the user.
+     */
+    updateWorkingTimeDuration() {
+        const seconds = Math.round(this.regularExamWorkingTime * (1.0 + this.workingTimeFormValues.percent / 100));
+        this.setWorkingTimeDuration(seconds);
+    }
+
+    /**
+     * Updates the hours, minutes, and seconds values of the form.
+     * @param seconds the total number of seconds of working time.
+     * @private
+     */
+    private setWorkingTimeDuration(seconds: number) {
+        const workingTime = this.artemisDurationFromSecondsPipe.secondsToDuration(seconds);
+        this.workingTimeFormValues.hours = workingTime.days * 24 + workingTime.hours;
+        this.workingTimeFormValues.minutes = workingTime.minutes;
+        this.workingTimeFormValues.seconds = workingTime.seconds;
     }
 
-    examIsVisible(): boolean {
+    /**
+     * Uses the current durations saved in the form to update the extension percent value.
+     */
+    updateWorkingTimePercent() {
+        this.workingTimeFormValues.percent = this.getWorkingTimePercentDifference();
+    }
+
+    /**
+     * Calculates how many seconds the currently set working time has in total.
+     * @private
+     */
+    private getWorkingTimeSeconds(): number {
+        return this.workingTimeFormValues.hours * 3600 + this.workingTimeFormValues.minutes * 60 + this.workingTimeFormValues.seconds;
+    }
+
+    /**
+     * Calculates the difference in whole percent between the regular working time and the currently set individual working time.
+     *
+     * E.g., with a regular time of "1h" and a current value of "1h30min" returns 50.
+     * @private
+     */
+    private getWorkingTimePercentDifference(): number {
+        return Math.round((this.getWorkingTimeSeconds() / this.regularExamWorkingTime - 1.0) * 100);
```

Comment:
Because I want to get the difference to the regular working time (≙ 100%). I.e., a student exam with an individual working time identical to the regular working time has 1.0 times the regular working time but an extension of 0.

Reply:
I see, do you think it would make sense to add brackets around the division? It's mathematically not necessary but I think it gets easier to read.

Reply:
ESlint doesn’t like unnecessary parentheses and would complain if I added them.

Reply:
Eslint 🙄 
Thanks for checking :D